### ISSUE TITLE
Variable viewer

### ISSUE THREAD

**Issue by Ladvien** [Ikuyadeu/vscode-R#15]
I'd really like to see this extension have an R variable viewer.  This is the only reason I'm stuck using RStudio or Visual Studio, as they have integrated variable viewers.

I think I could hack it out, but I'm having problems finding a way to get information from the Terminal in the current vscode API.

[Get Line from Terminal](https://stackoverflow.com/questions/44990488/get-line-from-terminal)

Had any thoughts of implementing this feature?

**Comment by Ikuyadeu** [comment]
Thank you contributing offer @Ladvien. 
I could't understand your question yet.
But If you want to call R terminal I think you can use this
https://github.com/Ikuyadeu/vscode-R/blob/master/src/extension.ts#L89-L91

**Comment by Ladvien** [comment]
I'm sorry for the confusion.  I'm asking if there is a function which would allow us to read the text in the terminal.  

In the [vscode API for the Terminal](https://code.visualstudio.com/docs/extensionAPI/vscode-api#_a-nameterminalaspan-classcodeitem-id1015terminalspan) object has `send.text()`, but I'd also need `read.text()`.  Or even better, an event like `term.onTextChange(text)`

For example, if I had a way to read the Terminal:
```
rTerm.sendText("setwd('/User/users/Bob/Data');
terminalReply = rTerm.readLine();
rVariable = parseForRVariable(terminalReply);
```

**Comment by Ikuyadeu** [comment]
OK, I don't know like as `readText`.
But in this line,
https://github.com/Ikuyadeu/vscode-R/blob/master/src/extension.ts#L183
I run R program and got output from `stdout`.
How do you feel this?

**Comment by Ladvien** [comment]
Ok, here's a proof of concept dataframe viewer.  It uses the Excel / CSV Extension [Excel Viewer](https://marketplace.visualstudio.com/items?itemName=GrapeCity.gc-excelviewer).

https://youtu.be/lHkAvZ6pvS0

I'm working through some issues.  Basically, it's super rough right now, just a proof of concept.  But it does the following:

Captures what is selected as the name of the dataframe, passes it to the RTerm to write as a CSV, polls the file size of the CSV to assure it is completely written, then opens the CSV, lastly, it uses the Excel Viewer `csv.preview` command.

There is a major problem due to this vscode issue:

https://github.com/Microsoft/vscode/issues/32608

Basically, if the dataframe is over 5mbs then it will not load the document. Sigh.  I deal with large data, so that's a deal breaker for me.  But hopefully the vscode team will increase the filesize limit. For that and other reasons I've not created a PR, but I'd like any critiques or input.

https://github.com/Ladvien/vscode-R

**Comment by Ikuyadeu** [comment]
@Ladvien Thank you good idea and I was excited by this movie!
I think you can get small data from `str` or `class` function.
Also `args` and `usage` function are good too.

```r
a <- c(1:5)
b <- 1

lapply(ls(), function(x) { 
  print(paste(x,
              class(get(x)),
              capture.output(str(get(x)), silent = T)[1],
        sep = ","))
})
```

**Comment by Ladvien** [comment]
@Ikuyadeu Interesting.  I'm not sure how to apply it to a variable viewer, though.

Let me know if you have any suggestions regarding cleaning up the code. I think the next release of vscode will include a fix for the file limit problem, so I'd like to put in a PR once it's fixed.

**Comment by Ladvien** [comment]
I went ahead and created a PR; I just added a notification of the 5 MB limit.  Let me know what needs to be cleaned up and I'll get to work.

**Comment by Ladvien** [comment]
It looks like the next version of VSCode will allow 50 MB files to be loaded.  But I've been thinking about how to handle files larger.

I work with datasets which can grow into the 2gb range, at least in a plain text format.  I'm curious if maybe we should consider a graceful fallback for files larger than 50 MBs. 

Perhaps calling this R function:

http://jeromyanglim.tumblr.com/post/33825729070/function-to-view-r-data-frame-in-spreadsheet

It would require users install the package [XLConnect](https://cran.r-project.org/web/packages/XLConnect/index.html), which is definitely a pain to manage.  But it would allow viewing of data larger than 50 MB.

I'd insert code to detect what viewers are available to the OS (LibreOffice, Excel, Numbers, etc.)

Thoughts?

**Comment by Ikuyadeu** [comment]
@Ladvien Your idea is interesting.
I think some people are still using old version VSCode yet, and your limit notification is better for default.
For add this function, some work will need,
1. add extension setting (e.g. `r.preview.larger.dataframe`: true or false(default is false))
2. install notification for XLConnect

**Comment by Ikuyadeu** [comment]
Fixed by #150
